src/sql/workbench/browser/modelComponents/table.component.ts

```diff
@@ -445,6 +445,8 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
 			}, index));
 
 			this._register(this._checkboxColumns.get(col.value).onChange((state) => {
+				this.data[state.row][state.column] = state.checked;
+				this.data = this.data; // We need this line to update the actual properties value that is exposed to the extensions.
```

Comment:
Should I just store this.data to a temp variable. Change the cell value and then assign the temp variable back to this.data. Even though that will be logically equivalent to this code, I feel it will be more understandable. 
```
const temp = this.data;
temp[state.row][state.column] = state.checked;
this.data = temp; 
```

Reply:
You could also just call `setPropertyFromUI` after changing the value. Or even make a new `sendPropertyChangedEvent` function on the base ComponentBase.